Misc/Adam.cpp

```diff
@@ -2736,14 +2866,34 @@ void AWToggleClickTrack(COMMAND_T*)
 
 void UpdateGridToolbar()
 {
-	static int cmds[3] =
+	static int cmds[14] =
 	{
 		NamedCommandLookup("_SWS_AWTOGGLETRIPLET"),
 		NamedCommandLookup("_SWS_AWTOGGLEDOTTED"),
+		NamedCommandLookup("_SWS_AWTOGGLESWING"),
 		NamedCommandLookup("_SWS_AWTOGGLECLICKTRACK"),
+		NamedCommandLookup("_SWS_SETGRID_PRESERVE_TYPE_4"),
+		NamedCommandLookup("_SWS_SETGRID_PRESERVE_TYPE_2"),
+		NamedCommandLookup("_SWS_SETGRID_PRESERVE_TYPE_1"),
+		NamedCommandLookup("_SWS_SETGRID_PRESERVE_TYPE_1_2"),
+		NamedCommandLookup("_SWS_SETGRID_PRESERVE_TYPE_1_4"),
+		NamedCommandLookup("_SWS_SETGRID_PRESERVE_TYPE_1_8"),
+		NamedCommandLookup("_SWS_SETGRID_PRESERVE_TYPE_1_16"),
+		NamedCommandLookup("_SWS_SETGRID_PRESERVE_TYPE_1_32"),
+		NamedCommandLookup("_SWS_SETGRID_PRESERVE_TYPE_1_64"),
+		NamedCommandLookup("_SWS_SETGRID_PRESERVE_TYPE_1_128")
 	};
-	for (int i = 0; i < 3; ++i)
+	for (int i = 0; i < 14; ++i)
```

Comment:
With C++11 now being enabled, range-based for loops can be used. Hard-coding the size of the array can be completely avoided:
```cpp
static const int cmds[] {
	NamedCommandLookup("_SWS_AWTOGGLETRIPLET"),
	NamedCommandLookup("_SWS_AWTOGGLEDOTTED"),
	// ...
};
```

```suggestion
	for (const int cmd : cmds)
```

Reply:
Ah nice to know SWS is full C++11 now :) Thank you

Reply:
Note that only syntax features are enabled. For compatibility with macOS 10.5, the C++11 standard library is not available.

Reply:
Yikes, this is why I always stuck with old C++ ways, cause I wasn't able to test directly on OSX